### PR TITLE
Update three pass services to enable live testing in sovereign clouds for multiple services

### PR DESCRIPTION
**Pull request by Luyunmt**
These changes contain fixes for the following services to be able to run live tests against AzureUsGovernment and AzureChina clouds. For some of the below services (core,extensions, search) the sovereign cloud live tests will be green with the changes. 
Core
Extensions
Search

@benbp @danieljurek @jongio

### CODE REVIEW

File: sdk/core/tests.yml [hunk 1]
```diff
@@ -4,3 +4,4 @@ extends:
   template: ../../eng/pipelines/templates/stages/archetype-sdk-tests.yml
   parameters:
     ServiceDirectory: core
+    Clouds: 'Public,Preview,UsGov,China'
```
Comment (pakrym) [inline sdk/core/tests.yml:7]: I don't think core has any live tests.

**Review by heaths — APPROVED** [review]

**Comment by benbp** [comment]
Looks good to me

**Comment by jongio** [comment]
@benbp  @danieljurek - Good to merge?

**Comment by benbp** [comment]
@jongio yes it looks like it. Do you need one of us to do it?

**Comment by jongio** [comment]
Yes please

**Comment by benbp** [comment]
@jongio I didn't think about merge access, sorry. Running quick live tests then I'll merge in

**Comment by tzhanl** [comment]
failures in China and USGov are caused by azure.identity version is 1.0.0. 1.2+ version should work.

**Comment by benbp** [comment]
@Luyunmt I'm seeing a few test failures for extensions and search in the pipelines I ran. Can you take a look?

**Comment by benbp** [comment]
@tzhanl is the search failure transient?

```
    "statusMessage": {
          "status": "Failed",
          "error": {
            "code": "BadGatewayConnection",
            "message": "The network connectivity issue encountered for 'Microsoft.Search'; cannot fulfill the request."
          }
        },

```

**Comment by tzhanl** [comment]
@benbp we had "re-run failed" for several times, that failure still show. We didn't see this failure in previous months.

**Pull request merged**